FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/MessageDetailsFragment.kt

```diff
@@ -1000,38 +999,41 @@ class MessageDetailsFragment : BaseFragment(), ProgressBehaviour, View.OnClickLi
 
     textViewPgpPublicKey.text = clipLargeText(block.keyDetails?.publicKey ?: block.content)
 
-    val existingPgpContact = block.existingPgpContact
+    val existingRecipientWithPubKeys = block.existingRecipientWithPubKeys
     val button = pubKeyView.findViewById<Button>(R.id.buttonKeyAction)
     if (button != null) {
-      if (existingPgpContact == null) {
-        initSaveContactButton(block, button)
-      } else if (TextUtils.isEmpty(existingPgpContact.fingerprint)
+      if (existingRecipientWithPubKeys == null) {
+        initSaveRecipientButton(block, button)
+      } else {
+        //todo-denbond7 temporary disable updating
+        button.gone()
+      }/*else if (existingRecipientWithPubKeys.publicKeys.firstOrNull()?.fingerprint.isNullOrEmpty()
         || keyDetails?.fingerprint?.equals(
-          existingPgpContact.fingerprint!!, ignoreCase = true
+          existingRecipientWithPubKeys.publicKeys.firstOrNull()?.fingerprint, ignoreCase = true
         ) == true
       ) {
         initUpdateContactButton(block, button)
       } else {
         initReplaceContactButton(block, button)
-      }
+      }*/
```

Comment:
Here it should be similar to other places:
 - if we already have this particular public key (by fingerprint) saved for this recipient, then we compare by last modified. If the one we already have is newer or same, we tell the user. If the received one is newer, we offer import (with "update")
 - if we don't have that pubkey for this recipient by fingerprint, offer import (with "save")

So it's not `firstOrNull` but we instead check if it's in the list, and then whether it's newer or not.

Reply:
Interesting. It seems I've disabled some functionality in `MessageDetailsFragment.kt` during migration and forgotten to fix it.(too many changes in this PR :( ) I have to fix that.

Reply:
Please file a separate issue for the future to add a test for this. Since it could be disabled without noticing, it's not covered by any test.